autogen/sr.rs

```diff
@@ -114,166 +74,70 @@ fn get_type_fn_name(name: &str) -> String {
     }
 }
 
-pub fn gen_sr_type_check(grammar: &structs::Grammar) -> String {
-    // Collect all types and their parameters in the following format:
-    //   (type-name: &str, Vec<(param-name: quote::Ident, param-type: quote::Ident)>)
-    let cases: Vec<_> = grammar
-        .instructions
+pub struct GeneratedOperands {
+    pub decoration: String,
+}
+
+pub fn gen_sr_operands(grammar_operand_kinds: &[structs::OperandKind]) -> GeneratedOperands {
```

Comment:
Can we name this function as `gen_sr_code_from_operand_kind_grammar` or something? "Operand" is already fairly overloaded; let's differentiate wherever possible.